glotaran/project/project.py

```diff
@@ -636,18 +643,33 @@ def create_scheme(
             The maximum number of function evaluations.
         clp_link_tolerance : float
             The CLP link tolerance.
+        data_lookup_overwrite: Mapping[str, LoadableDataset] | None
+            Allows to bypass the default dataset lookup in the project ``data`` folder and use a
+            different dataset for the optimization without changing the model. This especially
+            useful when working with preprocessed data. Defaults to ``None``.
 
         Returns
         -------
         Scheme
             The created scheme.
         """
+        if data_lookup_overwrite is None:
+            data_lookup_overwrite = {}
         loaded_model = self.load_model(model_name)
-        data = {dataset: self.load_data(dataset) for dataset in loaded_model.dataset}
+        data_lookup_overwrite = {
+            dataset_name: dataset_value
+            for dataset_name, dataset_value in data_lookup_overwrite.items()
+            if dataset_name in loaded_model.dataset
+        }
+        data = {
+            dataset_name: self.data[dataset_name]
+            for dataset_name in loaded_model.dataset
+            if dataset_name not in data_lookup_overwrite
+        }
         return Scheme(
             model=loaded_model,
             parameters=self.load_parameters(parameters_name),
-            data=data,
+            data=data | data_lookup_overwrite,
```

Comment:
```suggestion
            data=data | data_lookup_override
```

Reply:
Fixed in [5e7a47d](https://github.com/glotaran/pyglotaran/pull/1257/commits/5e7a47de52900b22acde581c9e934280d5e95975)